src/MongoDB/WriteConcern.c

```diff
@@ -406,6 +412,134 @@ static PHP_METHOD(WriteConcern, bsonSerialize)
 	convert_to_object(return_value);
 } /* }}} */
 
+/* {{{ proto string MongoDB\Driver\WriteConcern::serialize()
+*/
+static PHP_METHOD(WriteConcern, serialize)
+{
+	php_phongo_writeconcern_t* intern;
+	ZVAL_RETVAL_TYPE           retval;
+	php_serialize_data_t       var_hash;
+	smart_str                  buf = { 0 };
+	const char*                wtag;
+	int32_t                    w;
+	int32_t                    wtimeout;
+
+	intern = Z_WRITECONCERN_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	if (!intern->write_concern) {
+		return;
+	}
+
+	wtag     = mongoc_write_concern_get_wtag(intern->write_concern);
+	w        = mongoc_write_concern_get_w(intern->write_concern);
+	/* Note: PHP currently enforces that wimeoutMS is a 32-bit integer, so
+	 * casting will never truncate the value. This may change with PHPC-1411. */
+	wtimeout = (int32_t) mongoc_write_concern_get_wtimeout_int64(intern->write_concern);
+
+#if PHP_VERSION_ID >= 70000
+	array_init_size(&retval, 3);
+
+	if (wtag) {
+		ADD_ASSOC_STRING(&retval, "w", wtag);
+	} else if (mongoc_write_concern_get_wmajority(intern->write_concern)) {
+		ADD_ASSOC_STRING(&retval, "w", PHONGO_WRITE_CONCERN_W_MAJORITY);
+	} else if (w != MONGOC_WRITE_CONCERN_W_DEFAULT) {
+		ADD_ASSOC_LONG_EX(&retval, "w", w);
+	}
+
+	if (mongoc_write_concern_journal_is_set(intern->write_concern)) {
+		ADD_ASSOC_BOOL_EX(&retval, "j", mongoc_write_concern_get_journal(intern->write_concern));
+	}
+
+	if (wtimeout != 0) {
+		ADD_ASSOC_LONG_EX(&retval, "wtimeout", wtimeout);
```

Comment:
Related to my comment above, this will likely need to be changed to conditionally use `ADD_ASSOC_STRING` for values larger than `INT32_MAX`. Likewise below in the PHP 5.x `#else` case.

Reply:
Acknowledging that this will be addressed with #1042.